### PR TITLE
Hybrid ROS1/ROS2 bridge

### PR DESCRIPTION
**Pull request by fpasch**
This change is [Reviewable](https://reviewable.io/reviews/carla-simulator/ros-bridge/441)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -26,6 +26,53 @@ jobs:
       env:
         - ROS_DISTRO=noetic
         - PYTHON_SUFFIX="3.7"
+      addons:
```
Comment (hofbi) [inline .travis.yml:29]: All those additions in the travis file are probably in by accident. We already moved them to the github actions

**Review by hofbi — CHANGES_REQUESTED** [review]

**Comment by hofbi** [comment]
I guess we need to setup a separate pipeline for ROS 2 in the CI. In the current way it fails as for ROS 2 the docker images are different than for ROS 1

File: packaging/build-deb.sh [hunk 2]
```diff
@@ -5,7 +5,8 @@ set -e
 SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
 BUILD_DIR=${SCRIPT_DIR}/build/$(rosversion -d)
 WORKSPACE_DIR=${BUILD_DIR}/catkin_ws
-if [ "$(rosversion -d)" = "noetic" ]; then
+ROS_VERSION=$(rosversion -d)
+if [ "$ROS_VERSION" = "noetic" -o "$ROS_VERSION" = "foxy" ]; then
```
Comment (hofbi) [inline packaging/build-deb.sh:9]: I would rather test for not equal to "melodic". Then we don't need to change this for new ROS 2 versions

File: packaging/install_dependencies.sh [hunk 3]
```diff
@@ -2,12 +2,22 @@
 
 SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
 ROS_VERSION=$(rosversion -d)
-if [ "$ROS_VERSION" = "noetic" ]; then
+if [ "$ROS_VERSION" = "noetic" -o "$ROS_VERSION" = "foxy" ]; then
```
Comment (hofbi) [inline packaging/install_dependencies.sh:5]: I would rather test for not equal to "melodic". Then we don't need to change this for new ROS 2 versions

File: packaging/install_dependencies.sh [hunk 4]
```diff
@@ -2,12 +2,22 @@
...
     PYTHON_SUFFIX=3
 else
     PYTHON_SUFFIX=""
 fi
 
+if [ "$ROS_VERSION" = "foxy" ]; then
```
Comment (hofbi) [inline packaging/install_dependencies.sh:11]: Here I would also use not equal to melodic or noetic to be more robust against changes

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -35,20 +35,31 @@ jobs:
...
         packaging/install_dependencies.sh
     - name: Init Workspace
+      if: matrix.ros-version != 'foxy'
       run: |
         mkdir -p $GITHUB_WORKSPACE/../catkin_ws/src
         cd $GITHUB_WORKSPACE/../catkin_ws/src
         ln -s $GITHUB_WORKSPACE
         cd ..
         catkin init
     - name: Build, Test, Lint
```
Comment (hofbi) [inline .github/workflows/ci.yml:50]: Now I would name with `ROS 2: Build, Test, Lint

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -35,20 +35,31 @@ jobs:
...
+      if: matrix.ros-version == 'foxy'
       shell: bash
       run: |
-        source /opt/ros/${{ matrix.ros-version }}/setup.bash
+        source /opt/ros/$(rosversion -d)/setup.bash
+        colcon build --continue-on-error
+        colcon test && colcon test-result
+        source install/setup.bash
+    - name: Build, Test, Lint
```
Comment (hofbi) [inline .github/workflows/ci.yml:58]: Now I would name with `ROS 1: Build, Test, Lint

File: .github/workflows/ci.yml [hunk 7]
```diff
@@ -35,20 +35,31 @@ jobs:
...
       run: |
         sudo apt-get update && sudo apt-get install wget unzip -y
-        rm -rf carla_msgs && wget https://github.com/carla-simulator/ros-carla-msgs/archive/master.zip
-        unzip master.zip -d carla_msgs && rm master.zip
+        #TODO: replace ros2 with master
+        rm -rf carla_msgs && wget https://github.com/carla-simulator/ros-carla-msgs/archive/ros2.zip
+        unzip ros2.zip -d carla_msgs && rm ros2.zip
         packaging/install_dependencies.sh
     - name: Init Workspace
+      if: matrix.ros-version != 'foxy'
```
Comment (hofbi) [inline .github/workflows/ci.yml:43]: To avoid the doubled check for ROS 1, we could put the code of the init workspace step into the ROS 1 build step

**Review by hofbi — CHANGES_REQUESTED** [review]

**Review by fpasch — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/carla-simulator/ros-bridge/441)* status: 0 of 189 files reviewed, 1 unresolved discussion (waiting on @hofbi)

___
*[.travis.yml, line 29 at r1](https://reviewable.io/reviews/carla-simulator/ros-bridge/441#-MOloP9sxMVO_aGj6Zfu-r1-29:-MUyFY_s9NarH0n4Aoi1:b-896fix) ([raw file](https://github.com/carla-simulator/ros-bridge/blob/f1a85a57b3d5eb59f1721cb49244dac7185a83c2/.travis.yml#L29)):*
<details><summary><i>Previously, hofbi (Markus Hofbauer) wrote…</i></summary><blockquote>

All those additions in the travis file are probably in by accident. We already moved them to the github actions
</blockquote></details>

Done.

**Review by hofbi — APPROVED** [review]

**Review by joel-mb — APPROVED** [review]
> Reviewed 1 of 184 files at r1.
> *[Reviewable](https://reviewable.io/reviews/carla-simulator/ros-bridge/441)* status: 1 of 193 files reviewed, 1 unresolved discussion (waiting on @hofbi)

**Review by hofbi — APPROVED** [review]

**Review by joel-mb — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/carla-simulator/ros-bridge/441)* status: 1 of 193 files reviewed, 1 unresolved discussion (waiting on @hofbi)

**Review by joel-mb — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/carla-simulator/ros-bridge/441#-:-MWPFcZq45HsqgIBxYPc:bnfp4nl)* status: 1 of 193 files reviewed, 1 unresolved discussion (waiting on @hofbi)

**Pull request merged**